Document java plugin requirement

wtanaka opened the pull request:
Without it, the emitted error messages are confusing

wtanaka commented:
I signed it!

loosebazooka commented on README.md:
```diff
@@ -11,6 +11,8 @@ The plugin provides tasks for uploading, downloading, running and managing [Goog
 To use the App Engine plugin, include in your build script:
 
+    apply plugin: 'java'
+    apply plugin: 'war'
```
potentially, using this plugin, users can use an 'ear'

```
apply plugin: 'war' // or 'ear'
```

wtanaka replied:
updated

wtanaka committed:
Document java plugin requirement

Without it, the emitted error messages are confusing

See also
https://github.com/GoogleCloudPlatform/gradle-appengine-plugin/issues/16

loosebazooka commented on README.md:
```diff
@@ -11,6 +11,8 @@ The plugin provides tasks for uploading, downloading, running and managing [Goog
 
 To use the App Engine plugin, include in your build script:
 
```
actually since we're already in here can you use the ``` format on this?

```
- ```groovy
- apply plugin: 'java'
- apply plugin: 'war' // or 'ear'
- apply plugin: 'appengine'
- ```
```

wtanaka replied:
the prevailing other docs seem to say `gradle` instead of `groovy`, so I am guessing that's what's desired here since this is more specifically gradle code.

wtanaka commented:
https://github.com/GoogleCloudPlatform/gradle-appengine-plugin/pull/306
